cavern/src/main/java/org/opencadc/cavern/nodes/NodeUtil.java

```diff
@@ -702,14 +702,15 @@ public FileVisitResult postVisitDirectory(Path t, IOException ioe)
         try (DirectoryStream<Path> stream = Files.newDirectoryStream(np)) {
             for (Path file : stream) {
                 log.debug("[list] visit: " + file);
+                if (limit != null && (limit == 0 || limit == nodes.size())) {
```

Comment:
Now that this condition is before the addition of nodes, doesn't the check need to be limit + 1?

Anyhow, I would restore the old code and just check for limit == 0 before the directory stream is created, because we don't need to do that if we don't need any children.

Reply:
Updated code to bail out earlier as suggested when limit = 0.